### PR TITLE
Johnfreeman/daq deliverables issue105 standard image

### PR DESCRIPTION
**Pull request by jcfreeman2**

### CODE REVIEW

**Comment by jcfreeman2** [comment]
This PR is the followup to https://github.com/DUNE-DAQ/microservices/pull/61, which I closed after a discussion about Marco's microservice needs. Since that PR, I've updated the standard image to include Protobuf and the Python code it generates off of ers's `schema/ers/issue.proto` so that you can `import ers.issue_pb2` in your Python scripts. 

Instructions on how to test this PR are in https://github.com/DUNE-DAQ/microservices/tree/johnfreeman/daq-deliverables_issue105_standard_image#readme . Please note that while they refer to the `Alma9`-based microservices image, we have a fallback `Centos7`-based image, `ghcr.io/dune-daq/microservices_cc7:ec76`, which we can also test.

**Comment by mroda88** [comment]
Unfortunately `curl` and `git clone` don't work from the entrypoint so this is not enough for the new ERS microservice.

**Comment by plasorak** [comment]
For me, the base image should have the required functionality to get :
 - The base image should be relatively "generic". I.e. it should just provide specific packages that we commonly rely on (oracle, protobuf, git, etc.).
 - Microservices should start from that image (`FROM generic_daq_microservice:A9.v1.0.0`, for example at the top of their `Dockerfile`) and install the code that they need. So for example for the ERS streamer:
```
FROM generic_daq_microservice:A9.v1.0.0
ARGS ERSVERSION=dunedaq-v4.1.1

RUN git clone https://github.com/DUNE-DAQ/ers.git -b $ERSVERSION

# The following script could be created in the base image:
RUN create_pypkg_protobuf ers/schema/ers ersschema 
RUN cd ers && pip install . # or whatever is needed to install the ers streamer

ENTRYPOINT ["fancy-ers-streamer-db-writer"]
```

That way we ensure that no `git clone` gets used every time we start the image, and we still have a relatively generic image with common functionalities stored in the same place. This means we don't rely on network calls that can potentially fail, and on the sanity of the repo if someone accidentally modifies it destructively.

`druncschema` has an example of generating protobuf files and that can be installed with `pip install .` without interfering with the dunedaq-pyvenv: https://github.com/DUNE-DAQ/druncschema/tree/develop

File: standard/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+# elisa_client_api needed by the logbook microservice
+RUN git clone https://github.com/DUNE-DAQ/elisa_client_api.git && python3 -m pip install --upgrade setuptools && python3 -m pip install ./elisa_client_api 
+
+
+
+# protoc-24.3-linux-x86_64.zip is the latest zipfile available as of Sep-15-2023
+# See also https://grpc.io/docs/protoc-installation/#install-pre-compiled-binaries-any-os
+
+RUN curl -LO https://github.com/protocolbuffers/protobuf/releases/download/v24.3/protoc-24.3-linux-x86_64.zip && \
```
Comment (alessandrothea) [inline standard/Dockerfile]: isn't protobuf pip-installable?

**Review by alessandrothea — COMMENTED** [review]
See questions in the code

**Comment by jcfreeman2** [comment]
Just had a meeting with Ron, Bonnie and Pat; will merge into develop

**Pull request merged**